src/EditorFeatures/CSharpTest/UsePatternMatching/CSharpAsAndNullCheckTests.cs

```diff
@@ -559,5 +559,90 @@ static void M(ParameterSyntax parameter)
     }
 }");
         }
+
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsInlineTypeCheck)]
+        public async Task InlineTypeCheck3()
+        {
+            await TestInRegularAndScriptAsync(
+@"class C
+{
+    void M()
+    {
+        [|string|] x;
+        if ((x = o as string) != null)
```

Comment:
I personally don't think we need a new option here, since this isn't available on the command line yet.

@kuhlenh how do you feel about the preview question?